app/scenes/Settings/Security.tsx

```diff
@@ -220,8 +247,62 @@ function Security() {
           short
         />
       </SettingRow>
+
+      <SettingRow
+        label={t("Allowed Domains")}
+        name="allowedDomains"
+        description={t(
+          "The domains which should be allowed to create accounts. This applies to both SSO and Email logins. Changing this setting does not affect existing user accounts."
+        )}
+      >
+        <div>
+          {data.allowedDomains &&
+            data.allowedDomains.map((domain, index) => (
+              <Flex key={index} gap={4}>
+                <Input
+                  key={index}
+                  id={`allowedDomains${index}`}
+                  value={domain}
+                  placeholder="example.com"
+                  onChange={(ev) => {
+                    const newData = { ...data };
+
+                    newData.allowedDomains![index] = ev.currentTarget.value;
+                    setData(newData);
+                  }}
+                />
+                <Remove>
+                  <Tooltip tooltip={t("Remove Domain")} placement="top">
+                    <NudeButton onClick={() => handleRemoveDomain(index)}>
+                      <CloseIcon />
+                    </NudeButton>
+                  </Tooltip>
+                </Remove>
+              </Flex>
+            ))}
+        </div>
+
+        <Flex justify="space-between" gap={4}>
+          {!data.allowedDomains?.length ||
+          data.allowedDomains[data.allowedDomains.length - 1] !== "" ? (
+            <Button type="button" onClick={handleAddDomain} neutral>
+              <Trans>Add another</Trans>
+            </Button>
+          ) : (
+            <span />
+          )}
+
+          <Button type="button" onClick={handleChange}>
+            <Trans>Save Domains</Trans>
```

Comment:
```suggestion
            <Trans>Save changes</Trans>
```

I think your idea of only showing this button when there have been edits is a good one, we can combine that with a wording change here and it should be quite clear. Once saved the button can disappear again.